### ISSUE TITLE
Visualizations as preview windows

### ISSUE THREAD

**Issue by deboer-tim** [kui-shell/vscode-extension#1]
Kui has a number of visualizations for files that will also be in git and hence be in a VS Code user's workspace. Just like in the Markdown link below, it would be extremely cool if we could provide an action on the file types that we recognize and open a visualization window instead of or beside the source editor: https://code.visualstudio.com/docs/languages/markdown

Yes, I realize that this is skipping the command line (which is normally the root and launch point of Kui), but if we have the viz available then this would be a great way to reuse it and expand Kui's base while providing something very useful.

Hmm, you could do a similar thing outside of VS Code by opening just the vis if you type a filename or adding a simple command like "viz pipeline.yaml" or "kui pipeline.yaml".

**Comment by myan9** [comment]
Nice idea! Kui currently supports a synchronous edit and preview feature for apache-composer. We should start to generalize this feature for a broad/advanced usage. We currently have some distinction between `open` and `edit` command, e.g. `open a Markdown file` shows the rendered version, and `edit` show the source code. I think introducing a visualization command should be very useful!

**Comment by starpit** [comment]
i agree, this is great. we can leverage the vscode "command line" to good effect.

**Comment by starpit** [comment]
@myan9 given kui's "prescan" model, i.e. a model of all registered kui commands, we should be able to generate a vscode command registration list automatically, right?
